Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ContentSubgraph.php

```diff
@@ -181,9 +199,23 @@ public function findParentNode(NodeAggregateId $childNodeAggregateId): ?Node
     public function findNodeByPath(NodePath $path, NodeAggregateId $startingNodeAggregateId): ?Node
     {
         $currentNode = $this->findNodeById($startingNodeAggregateId);
-        if ($currentNode === null) {
-            return null;
-        }
+
+        return $currentNode
+            ? $this->findNodeByPathFromStartingNode($currentNode, $path)
+            : null;
+    }
+
+    public function findNodeByAbsolutePath(AbsoluteNodePath $path): ?Node
+    {
+        $currentNode = $this->findRootNodeByType($path->rootNodeTypeName);
+
+        return $currentNode
+            ? $this->findNodeByPathFromStartingNode($currentNode, $path->path)
+            : null;
+    }
+
+    private function findNodeByPathFromStartingNode(Node $currentNode, NodePath $path): ?Node
```

Comment:
Do we need to adjust this method to work with absolute path, where there is a filled up node name as node id?

Maybe we should consider making the Nodepath a collection of NodeName|NodeAggId

Reply:
I actually dislike to fill up a nodepath with node ids. It totally overqualifies the result.

How about defining an absolute node path like:

if there are all named nodes:

`/<Neos.Neos.Sites>/my-site/bla/blie`

but if `bla` has no node name, it currently would insert its id:

`/<Neos.Neos.Sites>/my-site/1234567543256746578/blie`

but to identify the node `blie` at the end, the last part is already enough. So how about we use the following notation? @nezaniel @mficzel ?

`/1234567543256746578/blie`


or maybe with a different syntax:

`/<1234567543256746578>/blie`

Reply:
I find it a bit confusing that we now have `findNodeByPath` and `findNodeByPathFromStartingNode`, both basically doing the same but with the arguments swapped.
But part of the confusion is maybe because this internal method is placed between the API methods. I would suggest to move it out of the way, down to the other private method.

> Do we need to adjust this method to work with absolute path, where there is a filled up node name as node id?

@mhsdesign This is a private method, why would we need to accept absolute paths?

> Maybe we should consider making the Nodepath a collection of NodeName|NodeAggId

Maybe it makes sense to solve this on the lowest level really, i.e. by changing the `Node` constructor from:
```php
public function __construct(
    // ...
    public readonly NodeAggregateId $nodeAggregateId,
    // ...
    public readonly ?NodeName $nodeName,
) {
}
```
to s.th. like:
```php
public NodeName $name;

public function __construct(
    // ...
    public readonly NodeAggregateId $nodeAggregateId,
    // ...
    ?NodeName $nodeName,
) {
    $this->name = $nodeName ?? NodeName::fromString($nodeAggregateId->value);
}
```

Reply:
@bwaidelich sorry my bad, i meant 

Do we need to adjust this method to work with paths, where there is a filled up node name as node id?

I actually still like my proposal `/<1234567543256746578>/blie`

Reply:
@mhsdesign `/1234567543256746578/blie` would mean to extend the concept of absolute pathes further and introduce new features. Would not recommend especially as we want to limit the use to specific cases.